Add rule for Point x: x y: y -> x@y

AngelHely opened the pull request:
This pull request propose a new rule for the creation of instance Point  #14302

AngelHely committed:
Add rule for Point x: x y: y -> x@y

This pull request propose a new rule for the creation of instance Point 
#14302

astares commented on src/General-Rules/RePointRule.class.st:
```diff
@@ -0,0 +1,46 @@
+RePointRule >> basicCheck: aNode [
+
+	aNode isMessage ifFalse: [ ^ false ].
+	^ #( #x:y: ) includes: aNode selector
```
What if I have my own custom class and send an x:y: message to it - then it would lament as well (although it is not Point where the message is sent to) 

So I think the rule should check additionally for Point being the receiver of the message, no?

jecisc replied:
I agree. And it would be nice to add a test on this exception

AngelHely committed:
improvement rule for Point x:x y:y -> x@y

this commit improve RePointRule class to check additionally for Point being the receiver of the message

AngelHely committed:
Merge branch 'pharo-project:Pharo12' into add_rule_point

jecisc reviewed (approved):
Nice thank you.

I'll not integrate it yet since this is a new feature and we are in feature freeze and the CI is down

jecisc commented:
Updating the base to P13 instead of P12
